application/libraries/PluginManager/PluginEvent.php

```diff
@@ -173,6 +173,20 @@ public function set($key, $value)
         return $this;
     }
 
+    /**
+     * Set mass key/value pair to be used by plugins hanlding this event.
+     *
+     * @param array $values
+     * @return \PluginEvent Fluent interface
+     */
+    public function setFromArray($values)
```

Comment:
Only used for test ?

Reply:
Right now, yes. New cases shall appear as it is available.

Reply:
We already have API issue with only a single function, but you add more ?

Have such function disable all core fix see https://github.com/LimeSurvey/LimeSurvey/pull/2753 or https://github.com/LimeSurvey/LimeSurvey/pull/2743 …

Reply:
I don't say there are conflict , clearly not **currently**. But in some month/year : there are potentially more API broke.

Why not, but i don't really find why adding this function is interesting. 

Maybe i'm wrong ?

Reply:
Don't agree. Still, already rolledback :)